io/plugins/eu.esdihumboldt.hale.io.haleconnect.ui/src/eu/esdihumboldt/hale/io/haleconnect/ui/projects/HaleConnectTarget.java

```diff
@@ -392,11 +393,20 @@ private void prefillTargetProject() {
 	 * Update the page state.
 	 */
 	protected void updateState() {
-		updateLoginStatus();
 		updateOverwriteWarning();
 
-		setValid(haleConnect.isLoggedIn() && (ownerUser.isEnabled() || ownerOrg.isEnabled())
-				&& (newProject.getSelection() || targetProject != null));
+		boolean valid = haleConnect.isLoggedIn();
+		// Creating or updating transformation projects is only possible
+		// if the current user has sufficient permissions to create
+		// projects either in the user or in an organization context
+		valid &= ownerUser.getEnabled() || ownerOrg.getEnabled();
+		if (!newProject.getSelection()) {
+			// If the option to update an existing project is selected,
+			// a target project must be selected before proceeding.
+			valid &= targetProject != null;
```

Comment:
Quick research shows that by using the `&` operator there is not support for short-circuiting so it maybe makes sense to switch to `= valid &&` instead?

Reply:
The lack of short-circuit support definitely makes a difference in case there are function calls involved that have side effects. While I don't think this was the case here, I've changed it for clarity. Thanks for the feedback!